### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by bimandas1**

### CODE REVIEW

File: form.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+    $fname = $lname = $fullname = "";
```
Comment (sandipta18) [inline form.php:2]: Follow either camel case or snake case naming convention for declaring variables.

File: form.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    $target_file;
+    $img_upload = false;
```
Comment (sandipta18) [inline form.php:4]: TRUE/FALSE should be in caps.

File: form.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    if(isset($_POST["submit"])){
```
Comment (sandipta18) [inline form.php:6]: Please follow coding standards. It should be like this ->
`if (condition) {`

File: form.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php
+    $fname = $lname = $fullname = "";
+    $target_file;
```
Comment (sandipta18) [inline form.php:3]: Please add comment regarding the purpose for this variable

File: form.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    $img_upload = false;
+
+    if(isset($_POST["submit"])){
+        $fname = $_POST["fname"];
+        $lname = $_POST["lname"];
+        $fullname = $fname . " " . $lname;
+
+        if(isset($_FILES["image"]) && $_FILES["image"]["error"] == UPLOAD_ERR_OK){
```
Comment (sandipta18) [inline form.php:11]: Please follow coding standards.

File: form.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            $target_file = "uploads/" . basename($_FILES["image"]["name"]);
+
+            if(false && move_uploaded_file($_FILES["image"]["tmp_name"], $target_file)){
+                $img_upload = true;
+            }else{
+                echo "<p style='color:red'> Unsuccessfull ! </p>";
+            }
```
Comment (sandipta18) [inline form.php:18]: Please follow proper coding standards. 
And we should not write html inside PHP.

File: index.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        <br> <br> Last Name : 
+        <input type="text" name="lname" value="<?php echo $lname ?>" placeholder="Enter your Last name">
+        <br> <br> Full Name : 
+        <input type="text" name="fullname" value="<?php echo $fullname ?>" placeholder="Full Name">
+        <br> <br> Image :
+        <input type="file" name="image">
+        <br> <br> 
+        <input type="submit" name="submit">
+    </form>
+    <br> <br>
```
Comment (sandipta18) [inline index.php:25]: Why we are using `<br>` tag here?

File: index.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    <?php
+        if($img_upload === true){
+            echo "Immage uploaded successfully.";
+            echo "<img src='$target_file' alt='Uploaded Image'> ";
```
Comment (sandipta18) [inline index.php:30]: We should not write html inside PHP.

File: index.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        <input type="text" name="fullname" value="<?php echo $fullname ?>" placeholder="Full Name">
+        <br> <br> Image :
+        <input type="file" name="image">
+        <br> <br> 
+        <input type="submit" name="submit">
+    </form>
+    <br> <br>
+
+    <?php
+        if($img_upload === true){
```
Comment (sandipta18) [inline index.php:28]: Please follow proper coding standards.

**Review by sandipta18 — CHANGES_REQUESTED** [review]
1. Add an empty line at the last of each file otherwise this red circle will show.
2. There are some PHP coding standards. Please go through that and follow those coding standards while writing PHP code.

**Comment by sandipta18** [comment]
As discussed the implementation for the task 
`Full name: User cannot enter a value in Full name field. It will be disabled by default. When the first name and last name fields are filled, this field outputs the sum of the above 2 fields.` is incorrect.
So please make necessary changes.

File: creds.php [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?php
+    $api_key = "6791558286ad4a92929083ab626e3b3c";
```
Comment (sandipta18) [inline creds.php:2]: Keep this file in gitignore

File: emailValidate.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?php
+    function isValidEmail($email) {
```
Comment (sandipta18) [inline emailValidate.php:2]: Function docblock missing

File: emailValidate.php [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        curl_setopt_array($ch, [
+            CURLOPT_URL => "https://emailvalidation.abstractapi.com/v1/?api_key=$api_key&email=$email",
+            CURLOPT_RETURNTRANSFER => TRUE,
+            CURLOPT_FOLLOWLOCATION => TRUE
+        ]);
+
+        $response = curl_exec($ch);
+
+        $data = json_decode($response, TRUE);
+        // var_dump($data);
```
Comment (sandipta18) [inline emailValidate.php:16]: Please remove commented out code.
Do the same in all the files.

File: form.php [hunk 13]
```diff
@@ -1,4 +1,6 @@
 <?php
+    include 'emailValidate.php';
+
     $firstName = $lastName = $marks_str = "";
```
Comment (sandipta18) [inline form.php:4]: Why are we mixing variable naming convention here? Let's either keep it `snake_case` or  `camelCase

**Review by sandipta18 — CHANGES_REQUESTED** [review]

File: form.php (hunk 3 above)
Comment (bimandas1) [reply form.php:19]: Solved

File: emailValidate.php (hunk 12 above)
Comment (bimandas1) [reply emailValidate.php:16]: solved.

File: form.php (hunk 13 above)
Comment (bimandas1) [reply form.php:4]: solved

**Comment by sandipta18** [comment]
No need to  push packages/libraries over github. Please keep the fpdf directory in gitignore.

File: creds.php (hunk 10 above)
Comment (bimandas1) [reply creds.php:2]: solved.

**Pull request closed without merging**